frontend/app/src/utils/indexed-db.ts

```diff
@@ -0,0 +1,78 @@
+export default class IndexedDb {
+  dbName: string = '';
+  dbVersion: number = 0;
+  store: string = '';
+
+  constructor(dbName: string, dbVersion: number, store: string) {
+    this.dbName = dbName;
+    this.dbVersion = dbVersion;
+    this.store = store;
+  }
+
+  get db(): Promise<IDBDatabase> | null {
+    return new Promise((resolve, reject) => {
+      if (!window.indexedDB) {
+        reject('Unsupported indexedDB');
+      }
+
+      const request = (
+        window.indexedDB ||
+        (window as any).mozIndexedDB ||
+        (window as any).webkitIndexedDB
+      ).open(this.dbName, this.dbVersion);
+
+      request.onsuccess = e => {
+        resolve((e.target as any).result);
+      };
+      request.onerror = e => {
+        reject(e);
+      };
+      request.onupgradeneeded = e => {
+        const db = (e.target as any).result;
+        if (!db.objectStoreNames.contains(this.store)) {
+          db.createObjectStore(this.store, {
+            keyPath: 'id',
+            autoIncrement: true
+          });
+        }
+      };
+
+      request.onblocked = e => {
+        reject(e);
+      };
+    });
+  }
+
+  async add(data: any, callback: Function = () => {}) {
```

Comment:
This function will continue writing logs to the indexeddb but I can't see any data eviction. Can we add a limit either temporal *(e.g 24-48 hours)* or quantitative *(let's say 50000 entries)* after which any older entries will be evicted/removed when a new entry is inserted.   

I am not sure about what limits indexeddb has but is there any chance that as it grows it will start degrading performance? I remember hearing something along the lines. In any case we should make sure that the entries stored don't keep increasing forever.

Reply:
I decided to limit the entry to 50k. 
For estimation, 10 rows of data is 1775 bytes
That means 50k data is should be below 10MB.

But I still need to handle an exception, if the limit comes from the user hard disk free space.

Reply:
50k entries at around ~10MB sounds good